extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -81,8 +88,34 @@ public List<InterceptorBinding> getAdditionalBindings() {
     }
 
     @BuildStep
-    void transformWithSpan(
-            BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+    void transformWithSpan(BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+
+        // Transform deprecated annotation into new one
+        annotationsTransformer.produce((new AnnotationsTransformerBuildItem(context -> {
+
+            AnnotationTarget target = context.getTarget();
+
+            if (target.kind().equals(AnnotationTarget.Kind.METHOD)) {
```

Comment:
Instead of this check, please override the dedicated method on `AnnotationTransformer`, see https://github.com/brunobat/quarkus/blob/10e8beea570f72ec4471aaf89bddce75a31e372b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationsTransformer.java#L26-L34

Reply:
Will do